src/rate.rs

```diff
@@ -77,6 +77,30 @@ const DQP_Q57: &[i64; FRAME_NSUBTYPES] = &[
   (2.0 * (33_810_170.0 / 86_043_287.0) * (1i64 << 57) as f64) as i64,
 ];
 
+// For 8-bit-depth inter frames, log_q_y is derived from log_target_q with a
+//  linear model:
+//  log_q_y = log_target_q + (log_target_q >> 32) * Q_MODEL_MUL + Q_MODEL_ADD
+// Derivation of the linear models:
+//  https://github.com/YaLTeR/rav1e/blob/45629334536861fc5b6f601e42fd4799d5c46c46/tools/analyse-vstats.ipynb
+#[rustfmt::skip]
+const Q_MODEL_ADD: [i64; 3] = [
+  // 4:2:0
+  -0x24_4FE7_ECB3_DD90,
+  // 4:2:2
+  -0x37_41DA_38AD_0924,
+  // 4:4:4
+  -0x70_83BD_A626_311C,
+];
+#[rustfmt::skip]
+const Q_MODEL_MUL: [i64; 3] = [
+  // 4:2:0
+  0x8A0_50DD,
+  // 4:2:2
+  0x887_7666,
+  // 4:4:4
+  0x8D4_A712,
+];
```

Comment:
We may add entries for `ChromaSampling::Cs400` and remove its exclusion below.
```suggestion
const Q_MODEL_ADD: [i64; 4] = [
  // 4:2:0
  -0x24_4FE7_ECB3_DD90,
  // 4:2:2
  -0x37_41DA_38AD_0924,
  // 4:4:4
  -0x70_83BD_A626_311C,
  // 4:0:0
  0,
];
#[rustfmt::skip]
const Q_MODEL_MUL: [i64; 4] = [
  // 4:2:0
  0x8A0_50DD,
  // 4:2:2
  0x887_7666,
  // 4:4:4
  0x8D4_A712,
  // 4:0:0
  0,
];
```